### DISCUSSION TITLE
CUDA role in Memory Efficient Attention

### DISCUSSION THREAD

**Discussion by BillyGun27** [facebookresearch/xformers#689] [Q&A] [answered]
I am wondering why do we need NVIDIA GPUs with compute capability above 6.0 (P100+) for Memory-efficient attention? 
Does this function require a specific CUDA operation that is not exist in the one with lower capability?

**Comment by danthe3rd** [comment, marked as answer]
Hi,
Memory-efficient attention has been tested on 6.0+, but should work with 5.0+. This kernel relies on [CUTLASS](https://github.com/NVIDIA/cutlass/tree/2.11) which might not work below 5.0.
If you have an older GPU, it might work if you build from source (and with some luck), but I don't recommend that, and we won't support it if you have questions or issues.
